proposals/Symbiosis Protocol.md

```diff
@@ -0,0 +1,393 @@
+
+### Project definition
+##### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+
+1. Specificy [symbiosis wire protocol](#Symbiosys-protocol).
+2. Implement domain service that host announces (will figure out the right one by talking to collabs).
+3. Implement permission handling flow on the host.
+4. Make symbiont leverage announced service through the protocol.
+5. Collaborate with teams to validate design with differnt use cases.
+6. Take learnings and implement zeroconfig service discovery.
+
+##### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+
+- Have at least one collab adopt symbiosis protocol.
+- IPFS Desktop providing services that enhances in-browser node capabilities.
+
+#####  What does success look like?
+_Success means impact. How will we know we did the right thing?_
+
+- Collabs are adopting a protocol to extend a stack instead of forking it.
+
+<!--
+Provide success criteria. These might include particular metrics, desired changes in the types of bug reports being filed, desired changes in qualitative user feedback (measured via surveys, etc), etc.
+-->
+
+##### Counterpoints &amp; pre-mortem
+_Why might this project be lower impact than expected? How could this project fail to complete, or fail to be successful?_
+
+- It does not adequately solve problems of our users.
+- Solution is to generic to be practical.
+- Service discovery is not reliable and still requires node configuration.
+
+##### Alternatives
+_How might this project’s intent be realized in other ways (other than this project proposal)? What other potential solutions can address the same need?_
+
+- Implement permission and authentification layer to existing IPFS HTTP API.
+- Allow users to extend HTTP API with domain specific routes.
+
+##### Dependencies/prerequisites
+<!--List any other projects that are dependencies/prerequisites for this project that is being pitched.-->
+
+- none
+
+##### Future opportunities
+<!--What future projects/opportunities could this project enable?-->
+
+- IPFS in serverless compute.
+- IPFS node sharing across applications.
+- Domain specific protocol extensions as services (e.g. replication optimization based on DAG structure knowledge like Textile does)
+
+### Required resources
+
+##### Effort estimate
+<!--T-shirt size rating of the size of the project. If the project might require external collaborators/teams, please note in the roles/skills section below). 
+For a team of 3-5 people with the appropriate skills:
+- Small, 1-2 weeks
+- Medium, 3-5 weeks
+- Large, 6-10 weeks
+- XLarge, >10 weeks
+Describe any choices and uncertainty in this scope estimate. (E.g. Uncertainty in the scope until design work is complete, low uncertainty in execution thereafter.)
+-->
+
+- Large
+- Uncertainty on discovery implementation until several services are implemented.
```

Comment:
We will probably need to rely on manual configuration first, even though we can add some defaults like trying default IPFS Desktop address port pair